### PR TITLE
upgrading to python 3.12

### PR DESCRIPTION
**Pull request by hriprsd**
- upgrading to python 3.12
- solves #1285 
- python 3.12 used across project in the notified areas as in the issue. 
- updated changelog as well.

### CODE REVIEW

**Comment by hriprsd** [comment]
Also, could you please add `hacktoberfest-accepted` label to my PR please? would be a huge help! Thanks in advance!

**Comment by hriprsd** [comment]
Please do let me know if I have missed something/ any more changes are needed. Thanks in advance! 😄

**Comment by hriprsd** [comment]
Hi. Hoping this will be counted towards my hacktober contribution. P.s would need the label `hacktoberfest-accepted` added to my PR or `hacktober` topic to the repo please. 

Aplogies for the follow ups, but within October would be great
 🙂 

Thanks in advance!

**Comment by jefftriplett** [comment]
Thank you for your contribution. We'll review this week and add the tag after we accept it. Juggling day jobs but we will not forget to add it :)

File: pyproject.toml [hunk 1]
```diff
@@ -63,7 +63,7 @@ line-length = 88
 dummy-variable-rgx = "^(_+|(_+[a-zA-Z0-9_]*[a-zA-Z0-9]+?))$"
 
 # Assume Python 3.11
-target-version = "py311"
```
Comment (saadmk11) [inline pyproject.toml:66]: please update the comment above this line (`# Assume Python 3.11`)

**Review by saadmk11 — CHANGES_REQUESTED** [review]
Awesome work here 💯  This is looking great. 👍 

Seems like some tests are failing because python 3.12 removed `distutils` (`ModuleNotFoundError: No module named 'distutils'`), I think we can fix this by replacing `distutils` with the `packaging` package which we already install.

We may also want to re-compile the `requirements.txt` file with the new python version.
https://github.com/djangopackages/djangopackages/blob/c5fa68de85aa0c7e055ac4eac6090cf5f37d36ad/requirements.txt#L2

**Comment by hriprsd** [comment]
> Awesome work here 💯 This is looking great. 👍
> 
> Seems like some tests are failing because python 3.12 removed `distutils` (`ModuleNotFoundError: No module named 'distutils'`), I think we can fix this by replacing `distutils` with the `packaging` package which we already install.
> 
> We may also want to re-compile the `requirements.txt` file with the new python version.
> 
> https://github.com/djangopackages/djangopackages/blob/c5fa68de85aa0c7e055ac4eac6090cf5f37d36ad/requirements.txt#L2

Fixed. 😄

**Comment by jefftriplett** [comment]
TIL this, merged the PR for who only knows what reasons... 

```
$ git push origin hriprsd/main:main
```

**Comment by jefftriplett** [comment]
@hriprsd thank you for the contribution

@saadmk11 @ryancheley This is fixed in the main, and I found a package that replaces our usage of LooseVersion.

**Pull request merged**